File: Samples/CslaLight/vb/Rolodex/Business/Server/BusinessClasses/RanksFactory.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports Csla
Imports Csla.Security
Imports Csla.Core
Imports Csla.Serialization
Imports Csla.Silverlight
Imports Csla.Validation

#If(Not SILVERLIGHT) Then
Imports System.Data.SqlClient
Imports Rolodex.Business.Data
#End If

Namespace Rolodex.Business.BusinessClasses
  Public Class RanksFactory

#If(Not SILVERLIGHT) Then

	Public Function FetchRanks() As Ranks
	  Dim returnValue As New Ranks()
	  returnValue.RaiseListChangedEvents = False
	  returnValue.SetReadOnlyFlag(False)
	  returnValue.Add(New Ranks.NameValuePair(0, ""))

	  Using connection As New SqlConnection(DataConnection.ConnectionString)
		connection.Open()
		Using command As New SqlCommand("RanksSelect", connection)
		  command.CommandType = System.Data.CommandType.StoredProcedure
		  Using reader As New Csla.Data.SafeDataReader(command.ExecuteReader())
			Do While reader.Read()
			  returnValue.Add(New Ranks.NameValuePair(reader.GetInt32("RankID"), reader.GetString("Rank")))
			Loop
		  End Using
		End Using
		connection.Close()
	  End Using

	  returnValue.SetReadOnlyFlag(True)
	  returnValue.RaiseListChangedEvents = True
	  Return returnValue
	End Function
#End If
  End Class
End Namespace

File: Samples/CslaLight/cs/Rolodex/Business/Server/BusinessClasses/RanksFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Csla;
using Csla.Security;
using Csla.Core;
using Csla.Serialization;
using Csla.Silverlight;
using Csla.Validation;

#if!SILVERLIGHT
using System.Data.SqlClient;
using Rolodex.Business.Data;
#endif

namespace Rolodex.Business.BusinessClasses
{
  public class RanksFactory
  {

#if!SILVERLIGHT

    public Ranks FetchRanks()
    {
      Ranks returnValue = new Ranks();
      returnValue.RaiseListChangedEvents = false;
      returnValue.SetReadOnlyFlag(false);
      returnValue.Add(new Ranks.NameValuePair(0, ""));

      using (SqlConnection connection = new SqlConnection(DataConnection.ConnectionString))
      {
        connection.Open();
        using (SqlCommand command = new SqlCommand("RanksSelect", connection))
        {
          command.CommandType = System.Data.CommandType.StoredProcedure;
          using (Csla.Data.SafeDataReader reader = new Csla.Data.SafeDataReader(command.ExecuteReader()))
          {
            while (reader.Read())
            {
              returnValue.Add(new Ranks.NameValuePair(reader.GetInt32("RankID"), reader.GetString("Rank")));
            }
          }
        }
        connection.Close();
      }

      returnValue.SetReadOnlyFlag(true);
      returnValue.RaiseListChangedEvents = true;
      return returnValue;
    }
#endif
  }
}
